resources/repo.rb

```diff
@@ -0,0 +1,116 @@
+# To learn more about Custom Resources, see https://docs.chef.io/custom_resources.html
+#
+# Author:: Corey Hemminger (<hemminger@hotmail.com>)
+# Cookbook:: yum-mysql-community
+# Resource:: repo
+#
+# Copyright:: 2020, Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+resource_name 'yum_mysql_community_repo'
+
+description 'Configure yum repo for MySQL Community edition'
+
+property :version, String,
+         default: '8.0',
+         description: 'MySQL version to configure'
+
+property :gpgcheck, [true, false],
+         default: true,
+         description: 'Enable or disable GPG checks'
+
+property :mysql_community_server, [true, false],
+         default: true,
+         description: 'Enable or disable mysql community server repo'
+
+property :mysql_connectors_community, [true, false],
+         default: true,
+         description: 'Enable or disable mysql connectors community repo'
+
+property :mysql_tools_community, [true, false],
+         default: true,
+         description: 'Enable or disable mysql tools community repo'
+
+property :mysql_tools_preview, [true, false],
+         default: false,
+         description: 'Enable or disable mysql tools repo'
+
+property :mysql_cluster_community, [true, false],
+         default: false,
+         description: 'Enable or disable GPG checks mysql cluster community repo'
+
+action :create do
+  description 'Create MySQL Community Repo file'
+  os = case node['platform']
```

Comment:
You're better off using platform_family here and just doing rhel/amazon

Reply:
is fedora part of rhel platform_family?

Reply:
Nope fedora has its own platform family because it's oddly different

Reply:
made suggested change in latest commit